### PR TITLE
Adjust style guide (loops) and add unreachable macro

### PR DESCRIPTION
**Pull request by sea212**
### What does it do?
- Update style guide: Loop handling and unreachable code handling
- Add `unreachable_non_terminating` macro

### CODE REVIEW

File: primitives/src/macros.rs
Comment (Chralt98) [inline primitives/src/macros.rs:1]: I would have expected a `mod macros;` here  https://github.com/zeitgeistpm/zeitgeist/blob/fca15ab8cce1e9faa79f8292bdc3babe4e65a1e0/primitives/src/lib.rs#L23-L32

**Review by Chralt98 — APPROVED** [review]

File: primitives/src/macros.rs
Comment (sea212) [reply primitives/src/macros.rs:1]: https://github.com/zeitgeistpm/zeitgeist/pull/1252/commits/35b398f99a8f8079f3dbebc66cfa400e02664cac

File: docs/STYLE_GUIDE.md [hunk 1]
```diff
@@ -156,3 +156,17 @@ duplicating documentation.
...
+// In code logic
+zeitgeist_primitives::unreachable_non_terminating!(condition, log_target, message)
+```
+
+```rust
+// In test
+#[test]
+#[should_panic(expected = message)]
+// Cause assertion
+```
```
Comment (maltekliemann) [inline docs/STYLE_GUIDE.md:172]:
````suggestion
  ```rust
  // In code logic
  zeitgeist_primitives::unreachable_non_terminating!(condition, log_target, message)
  ```
  
  ```rust
  // In test
  #[test]
  #[should_panic(expected = message)]
  // Cause assertion
  ```
````

File: primitives/src/macros.rs
Comment (maltekliemann) [reply primitives/src/macros.rs:1]: Actually, we have a whole `macros` crate in our project root.

Comment (sea212) [reply primitives/src/macros.rs:1]: https://github.com/zeitgeistpm/zeitgeist/pull/1252/commits/1d64036c27c179e18ce6b4229599a65c0c59639e

**Review by maltekliemann — APPROVED** [review]

**Review by Chralt98 — APPROVED** [review]

**Pull request merged**